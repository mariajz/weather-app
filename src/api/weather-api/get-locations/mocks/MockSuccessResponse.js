export const mockSuccessResponse = [
    {
        id: 1120673,
        name: 'Kochi',
        region: 'Kerala',
        country: 'India',
        lat: 9.97,
        lon: 76.23,
        url: 'kochi-kerala-india',
    },
    {
        id: 3125114,
        name: 'Kochi-Shi',
        region: 'Ko¬çchi',
        country: 'Japan',
        lat: 33.56,
        lon: 133.53,
        url: 'kochi-shi-kochi-japan',
    },
    {
        id: 1120674,
        name: 'Kochinda',
        region: 'Orissa',
        country: 'India',
        lat: 21.73,
        lon: 84.35,
        url: 'kochinda-orissa-india',
    },
];
