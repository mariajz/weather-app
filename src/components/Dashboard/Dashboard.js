import React from 'react';
import { SafeAreaView } from 'react-native';
import { DashboardWrapper } from './Dashboard.style';
import SearchCitySection from '../search-city-section';

const Dashboard = () => {
    return (
        <SafeAreaView>
            <DashboardWrapper>
                <SearchCitySection />
            </DashboardWrapper>
        </SafeAreaView>
    );
};

export default Dashboard;
