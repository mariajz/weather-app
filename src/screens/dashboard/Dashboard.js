import React from 'react';
import { SafeAreaView } from 'react-native';
import SearchCitySection from '../../components/search-city-section';
import ForecastSection from '../forecast';
import { DashboardWrapper } from './Dashboard.style';

const Dashboard = ({ route }) => {
    const forecastSectiondata = route.params.data;
    return (
        <SafeAreaView>
            <DashboardWrapper>
                <SearchCitySection />
                <ForecastSection data={forecastSectiondata} />
            </DashboardWrapper>
        </SafeAreaView>
    );
};

export default Dashboard;
