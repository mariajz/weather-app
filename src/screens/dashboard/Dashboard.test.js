import { render } from '@testing-library/react-native';
import React from 'react';
import { View as MockView } from 'react-native';
import { mockSuccessResponse } from '../../api/weather-api/current-weather/mocks/MockSuccessResponse';
import Dashboard from './Dashboard';

jest.mock('../../components/search-city-section', () => () => (
    <MockView>search city section</MockView>
));
jest.mock('../forecast', () => () => <MockView>forecast section</MockView>);

jest.mock('../../states/useLoading');

const defaultProps = {
    route: {
        params: {
            data: mockSuccessResponse,
        },
    },
};
describe('Dashboard', () => {
    it('should render Dashboard', () => {
        const container = render(<Dashboard {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
