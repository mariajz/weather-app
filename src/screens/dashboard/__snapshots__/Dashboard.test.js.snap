// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render Dashboard 1`] = `
<RCTSafeAreaView>
  <View
    style={
      {
        "posiiton": "relative",
      }
    }
  >
    <View>
      search city section
    </View>
    <View>
      forecast section
    </View>
  </View>
</RCTSafeAreaView>
`;
